KILTprotocol/type-definitions
# feat: add did rpc decorations

## Fixes https://github.com/KILTprotocol/ticket/issues/1953

It adds type augmentation for the runtime APIs (callable with `api.call.didApi...`), ~but it has on hold the generation of types for the DID RPC stuff, as decoding is currently failing when fetching data using a Typescript script~.

## Checklist

- [x] Add definitions for RPC endpoints
- [x] Update polkadot package once its released with support for `AccountId33` as in https://github.com/polkadot-js/api/pull/5197/files

## How to test

After https://github.com/KILTprotocol/type-definitions/pull/33 is merged into this, you can spin up a local instance of polkadotJS apps, where you replace the KILT type definition with a locally published version of this package (e.g. with `yalc`). You should now be able to see the pubilc credential RPCs among the possible RPC methods, and also the runtime calls for both the DID and the public credentials runtime APIs.

One screenshot for the RPC and one for the runtime calls attached.

<img width="1680" alt="Screenshot 2022-08-11 at 16 24 16" src="https://user-images.githubusercontent.com/6704504/184156647-7e21e488-1295-4514-8b3d-8ba16f99e618.png">
<img width="1680" alt="Screenshot 2022-08-11 at 16 25 30" src="https://user-images.githubusercontent.com/6704504/184156659-0a0071c0-82c0-4f53-b6de-a9cc3dd970e4.png">


## How to integrate into the SDK

For usage with the SDK, the `typegen` package must be used, by first pulling in the metadata from an up-to-date chain that has these features (e.g. from `develop`), then updating `packages/augment-api/src/interfaces/extraDefs/definitions.ts` to use the `rpc`, `types`, and `runtime` definitions by pulling them in from this package, and the run `build:types`. The result should be an up-to-date typescript definition of all RPC and `state_call` stuff.
